### PR TITLE
Add interfaces

### PR DESCRIPTION
**Pull request by Lunkers**
Adds typescript interfaces to the repo. The interfaces can be used to build IAF-compliant plugins.

### CODE REVIEW

File: interfaces/interfaces.d.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+interface TranscodeDispatcher {
+    encodeParams: any,
+    inputLocation: string;
+    outputDestination: string;
+    logger: winston.Logger;
+    dispatch(fileName: string): Promise<any>;
+
+}
+
+interface IafFileWathModule {
```
Comment (birme) [inline interfaces/interfaces.d.ts:26]: Typo / spelling mistake?

**Review by birme — CHANGES_REQUESTED** [review]
I was wondering whether we should only have the "public" IAF interfaces here. Perhaps the Uploader and TranscodeDispatcher interfaces are a bit more internal and could differ depending on what service you integrate with. Or what do you think?

File: interfaces/interfaces.d.ts (hunk 1 above)
Comment (Lunkers) [reply interfaces/interfaces.d.ts:26]: Yup, typo. Will fix.

**Comment by Lunkers** [comment]
Good point. So in that case we'd only really expose `IafUploadModule` and `IafFileWatchModule`, right?

**Comment by birme** [comment]
> Good point. So in that case we'd only really expose `IafUploadModule` and `IafFileWatchModule`, right?

Yes exactly

**Review by birme — APPROVED** [review]

**Pull request merged**